plugins/command-manager/src/main/java/com/wazuh/commandmanager/jobscheduler/SearchJob.java

```diff
@@ -0,0 +1,218 @@
+            int index = searchResponse.getHits().getHits().length - 1;
+            return searchResponse
+                .getHits()
+                .getHits()[index];
+        } catch (Exception e) {
+            return null;
+        }
+    }
+
+    public void handlePage(Client client, SearchResponse searchResponse) throws IOException, IllegalStateException {
+        SearchHits searchHits = searchResponse.getHits();
+        for (SearchHit hit : searchHits) {
+            updateStatusField(client, hit, Status.SENT);
+            commandHttpRequest(hit);
+        }
+    }
+
+    private static void commandHttpRequest(SearchHit hit) throws IOException {
+        XContentBuilder xContentBuilder = XContentFactory.jsonBuilder();
+        hit.toXContent(xContentBuilder, ToXContent.EMPTY_PARAMS);
+        HttpRestClientDemo.run("https://httpbin.org/post", xContentBuilder.toString());
+    }
+
+    @SuppressWarnings("unchecked")
+    private void updateStatusField(Client client, SearchHit hit, Status status ) throws IllegalStateException {
+        Map<String, Object> commandMap =
+            getNestedValue(hit.getSourceAsMap(), "command", Map.class);
+        commandMap.put(Command.STATUS, status);
+        hit.getSourceAsMap().put("command", commandMap);
+        IndexRequest indexRequest =
+            new IndexRequest()
+                .index(CommandManagerPlugin.COMMAND_MANAGER_INDEX_NAME)
+                .source(hit.getSourceAsMap())
+                .id(hit.getId());
+        client.index(indexRequest).actionGet(CommandManagerPlugin.DEFAULT_TIMEOUT_SECONDS * 1000);
+    }
+
+    public SearchResponse pitQuery(Client client, String index, Integer resultsPerPage, PointInTimeBuilder pointInTimeBuilder, Object[] searchAfter) throws IllegalStateException {
+        SearchRequest searchRequest = new SearchRequest(index);
+        TermQueryBuilder termQueryBuilder =
+            QueryBuilders.termQuery(Command.COMMAND + "." + Command.STATUS + ".keyword", Status.PENDING);
+        getSearchSourceBuilder()
+            .query(termQueryBuilder)
+            .size(resultsPerPage)
+            .trackTotalHits(true)
+            .timeout(TimeValue.timeValueSeconds(CommandManagerPlugin.DEFAULT_TIMEOUT_SECONDS))
+            .pointInTimeBuilder(pointInTimeBuilder);
+        if( getSearchSourceBuilder().sorts() == null ) {
+            getSearchSourceBuilder()
+                .sort(Command.COMMAND + "." + Command.ORDER_ID + ".keyword", SortOrder.ASC)
+                .sort(Command.COMMAND + "." + Command.TIMEOUT, SortOrder.ASC);
+        }
+        if (searchAfter != null) {
+            getSearchSourceBuilder().searchAfter(searchAfter);
+        }
+        searchRequest.source(getSearchSourceBuilder());
+        return client.search(searchRequest)
+            .actionGet(TimeValue.timeValueSeconds(CommandManagerPlugin.DEFAULT_TIMEOUT_SECONDS));
+    }
+
+    public Runnable searchJobRunnable(Client client, String index, Integer pageSize) {
```

Comment:
- Remove `index` from the method arguments (and others) using the `COMMAND_MANAGER_INDEX_NAME` constant instead.
- Remove `client` from the method arguments (and others), using a class attribute instead

```java
public SearchJob(Client client) {
    this.client = client;
}
```

Reply:
Javdocs

Reply:
Consider promoting the `SearchJob` class to `Runnable`, using this method as `public void run()`.